File: Arcade/Intro/javascript/avoidObstacles.js
function avoidObstacles(inputArray) {
    let k = true;
    for(let i = 2; ; i++) {
        for(let o of inputArray) {
            if(o % i == 0) {
                k = !k;
                break;
            }
        }
        if(k) return i;
        k = true;
    }
}

File: Arcade/Intro/Java/AvoidObstacles.java
final class AvoidObstacles {

	int avoidObstacles(int[] inputArray) {
	    boolean k = true;
	    for(int i = 2; ; i++) {
	        for(int j : inputArray) {
	            if(j % i == 0) {
	                k = !k;
	                break;
	            }
	        }
	        if(k) return i;
	        k = true;
	    }
	}
}